apps/charterafrica/src/components/Datasets/Datasets.js

```diff
@@ -0,0 +1,98 @@
+import { Section } from "@commons-ui/core";
+import { Box } from "@mui/material";
+import { useRouter } from "next/router";
+import React, { useState, useEffect } from "react";
+
+import DatasetCard from "./DatasetCard";
+import DatasetFilterBar from "./DatasetFilterBar";
+import useDatasets from "./useDatasets";
+
+import NextPrevPagination from "@/charterafrica/components/NextPrevPagination";
+import queryString from "@/charterafrica/utils/datasets/queryString";
+
+function Datasets({
+  sx,
+  data: originalDatasets,
+  tags = [],
+  countries = [],
+  count: originalCount,
+}) {
+  const pageSize = 10;
+  const [page, setPage] = useState(1);
+  const [q, setQ] = useState("");
+  const [sort, setSort] = useState("");
+  const [selectedCountries, setSelectedCountries] = useState([]);
+  const [selectedTags, setSelectedTags] = useState([]);
+  const [totalPages, setTotalPages] = useState(
+    Math.ceil(originalCount / pageSize)
+  );
+  const [datasets, setDatasets] = useState(originalDatasets || []);
+  const router = useRouter();
+  const { asPath } = router;
+
+  const handleChangePage = (_, value) => setPage(value);
+  const handleChangeQ = (_, value) => setQ(value);
+  const handleChangeSort = (_, value) => setSort(value);
+  const handleChangeCountries = (_, value) => setSelectedCountries(value);
+  const handleChangeTags = (_, value) => setSelectedTags(value);
+
+  const query = queryString({
+    page,
+    q,
+    sort,
+    countries: selectedCountries,
+    tags: selectedTags,
+  });
+
+  useEffect(() => {
+    const pathname = asPath.split("?")[0];
+    router.push({ pathname, query }, undefined, {
+      scroll: false,
+      shallow: true,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [query]);
+
+  const { data, isLoading } = useDatasets({
+    q,
+    sort,
+    page,
+    pageSize,
+    countries: selectedCountries,
+    tags: selectedTags,
+  });
+
+  useEffect(() => {
+    if (!isLoading) {
+      const { datasets: filteredDatasets, count } = data || {};
+      setDatasets(filteredDatasets);
+      setTotalPages(Math.ceil(count / pageSize));
+    }
+  }, [data, isLoading]);
```

Comment:
Show linear progress bar when `isLoading` (similar to how Articles component handle it)